src/sentry/integrations/jira/views/base.py

```diff
@@ -15,12 +17,33 @@ def get_response(self, context):
         add the requisite CSP headers before returning it.
         """
 
-        context["ac_js_src"] = "https://connect-cdn.atl-paas.net/all.js"
-        response = render_to_response(self.html_file, context, self.request)
         sources = [
             self.request.GET.get("xdm_e"),
             options.get("system.url-prefix"),
         ]
-        sources_string = " ".join(s for s in sources if s)  # Filter out None
-        response["Content-Security-Policy"] = f"frame-ancestors 'self' {sources_string}"
+
+        settings.CSP_FRAME_ANCESTORS = [
+            "'self'",
+        ] + [s for s in sources if s and ";" not in s]
+        settings.CSP_STYLE_SRC = [
+            # same as default (server.py)
+            "'self'",
+            "'unsafe-inline'",
+        ]
+
+        if settings.STATIC_FRONTEND_APP_URL.startswith("https://"):
+            origin = "/".join(settings.STATIC_FRONTEND_APP_URL.split("/")[0:3])
+            settings.CSP_STYLE_SRC.append(origin)
+
+        header = "Content-Security-Policy"
+        if getattr(settings, "CSP_REPORT_ONLY", False):
+            header += "-Report-Only"
+
+        middleware = CSPMiddleware()
+        middleware.process_request(self.request)  # adds nonce
+
+        context["ac_js_src"] = "https://connect-cdn.atl-paas.net/all.js"
+        response = render_to_response(self.html_file, context, self.request)
+
+        response[header] = middleware.build_policy(request=self.request, response=response)
```

Comment:
Would it be useful to add tests around the expected headers?

Reply:
Let me check how to do it. I see codecov is not happy